### PR TITLE
Added edx.user.logout Event

### PR DESCRIPTION
**Pull request by mumerfarooq000**
Story Link
https://edlyio.atlassian.net/browse/EDE-917

- added the event edx.user.logout in logout.py file.
when user successfully logged out than this event is triggered. 

Type of change
Please select the options that are relevant.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Documentation Change

### CODE REVIEW

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 1]
```diff
@@ -55,8 +57,19 @@ def dispatch(self, request, *args, **kwargs):
         # Get the list of authorized clients before we clear the session.
         self.oauth_client_ids = request.session.get(edx_oauth2_provider.constants.AUTHORIZED_CLIENTS_SESSION_KEY, [])
 
+        user = copy.copy(request.user)
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:60]: Tests are failing, this line needs to be changed. Maybe we need to check if user is authenticated.

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 2]
```diff
@@ -55,8 +57,19 @@ def dispatch(self, request, *args, **kwargs):
...
         logout(request)
 
+        if not request.user.username:
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:63]: Do we need this check?

**Review by danialmalik — CHANGES_REQUESTED** [review]

File: openedx/core/djangoapps/user_authn/views/logout.py (hunk 2 above)
Comment (mumerfarooq000) [reply openedx/core/djangoapps/user_authn/views/logout.py:63]: just to make sure the existing user is logged out.

File: openedx/core/djangoapps/user_authn/views/logout.py (hunk 1 above)
Comment (mumerfarooq000) [reply openedx/core/djangoapps/user_authn/views/logout.py:60]: yeah looking into this, I am not sure if it's breaking on this

**Comment by mumerfarooq000** [comment]
jenkins run python

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by edtecheco** [comment]
Your PR has finished running tests. There were no failures.

**Comment by mumerfarooq000** [comment]
@danialmalik the checks have passed please review now

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 3]
```diff
@@ -55,8 +58,23 @@ def dispatch(self, request, *args, **kwargs):
         # Get the list of authorized clients before we clear the session.
         self.oauth_client_ids = request.session.get(edx_oauth2_provider.constants.AUTHORIZED_CLIENTS_SESSION_KEY, [])
 
+        trigger_event = True
+        # if tests are giving Anonymous User objects
+        if not isinstance(request.user, AnonymousUser):
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:63]:
```suggestion
        if not request.user.is_anonymous:
```

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 4]
```diff
@@ -55,8 +58,23 @@ def dispatch(self, request, *args, **kwargs):
...
+                'email': request.POST.get('email'),
+                'username': request.user.username,
+                'user_id': request.user.id
+            }
+        else:
+            trigger_event = False
+
         logout(request)
 
+        if trigger_event and isinstance(request.user, AnonymousUser):
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:74]:
```suggestion
        if request.user.is_anonymous:
```

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 5]
```diff
@@ -55,8 +58,23 @@ def dispatch(self, request, *args, **kwargs):
...
+        trigger_event = True
+        # if tests are giving Anonymous User objects
+        if not isinstance(request.user, AnonymousUser):
+            event_data = {
+                'email': request.POST.get('email'),
+                'username': request.user.username,
+                'user_id': request.user.id
+            }
+        else:
+            trigger_event = False
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:70]:
```suggestion

```

File: openedx/core/djangoapps/user_authn/views/logout.py [hunk 6]
```diff
@@ -55,8 +58,23 @@ def dispatch(self, request, *args, **kwargs):
         # Get the list of authorized clients before we clear the session.
         self.oauth_client_ids = request.session.get(edx_oauth2_provider.constants.AUTHORIZED_CLIENTS_SESSION_KEY, [])
 
+        trigger_event = True
```
Comment (danialmalik) [inline openedx/core/djangoapps/user_authn/views/logout.py:61]:
```suggestion
```

**Review by danialmalik — APPROVED** [review]
Just some minor suggestions. Otherwise looks good.

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by mumerfarooq000** [comment]
jenkins run python

**Comment by edtecheco** [comment]
Your PR has finished running tests. There were no failures.

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by edtecheco** [comment]
Your PR has finished running tests. There were no failures.

**Pull request merged**